### PR TITLE
Create compiler types - great for DF compilers

### PR DESCRIPTION
**Pull request by giftedl**
I'm working on @df-js, and I needed a quick way for players, to instantly create locations. I wanted this to be on recode, because its the all in one mod, its great, as it has good features for regular coding.

### Why would we need it in recode
It would be great, and we already feature you, as you have (yes, i know, its from codeutilities) template sockets. They help export the template into, actual working DF code.

### The fixed problem
The up and big problem, is variables like locations, or even vectors. They can be incorrectly typed, which then you have to recompile, and then get it right. We want @df-js to be as quick as possible, and recompiling doesn't accomplish our goal. 

### What does it do?
You hold a location, type `/baseloc`, and it copy's a base64 string, which you would put inside your compilers code. Its really simple, and gets rid of non location variables.

### The future of other types
I am planning to make pull requests for other types, like vectors.

### My experience, and no kotlin???!?!
Well, first, I am pretty good at Java and have specifically made Bukkit plugins. I haven't really coded in Fabric, but it was easy enough for me to figure out. Well, I couldn't get kotlin because first (1) I have no experience with it and two (2) There were absolutely no examples on how to make commands in kotlin.

If your reading this, have a good day :heart:

### CODE REVIEW

**Comment by fallow64** [comment]
Perhaps just one single command (maybe something like `encode` or `getdata`) could be implemented for all types of values. If `baseloc` were to be added, then it just wouldn't be efficient to also add totally separate commands like `basevec` or `baseitem` when it could just all be encompassed into one.

Also, I'm commenting this here because I'm not too familiar with git, but this section:
```
if (tag != null) {
    if (tag.getString("hypercube:varitem") != "") {
        // ....
    } else {
        // error
    }
} else {
    // same error
}
```
Could just be condensed to:
```
if (tag != null && tag.getString("hypercube:varitem") != "") {
    // ...
} else {
    // error
}
```

**Comment by SamMan56** [comment]
why does this pr contain changes to gradle, gradlew, mod properties etc

File: src/main/resources/fabric_mod_json_template.txt
Comment (tk2217) [inline src/main/resources/fabric_mod_json_template.txt:1]: Don't mess with this stuff, adds unnecessary stuff to the pr.

File: gradle/wrapper/gradle-wrapper.jar
Comment (tk2217) [inline gradle/wrapper/gradle-wrapper.jar:1]: Same as below for these.

File: src/main/resources/fabric_mod_json_template.txt
Comment (giftedl) [reply src/main/resources/fabric_mod_json_template.txt:1]: oops, i had a issue with my gpu and i thought messing around with that stuff would fix it, i can fix that

**Comment by giftedl** [comment]
Alright im gonna close this and add the following

- [ ] Less messed up stuff
- [ ] Vector, and locations
I will tell you guys a pull request here when its done :)

**UPDATE** it is fully working look on #45

**Comment by GeorgeRNG** [comment]
@DeveloLongScript you might be interested in my separate mod, it is designed to allow complete replacements of how you code in df, such as external compilers. I have already have placing templates from an external script working, so I might be able to get your project to compile straight to df. It doesn't add anything not in recode, and should be compatible. I would be willing to add these features.

**Comment by DeveloLongScript** [comment]
> @DeveloLongScript you might be interested in my separate mod, it is designed to allow complete replacements of how you code in df, such as external compilers. I have already have placing templates from an external script working, so I might be able to get your project to compile straight to df. It doesn't add anything not in recode, and should be compatible. I would be willing to add these features.

Its okay, i can do them myself! thanks for asking though

**Comment by giftedl** [comment]
> @DeveloLongScript you might be interested in my separate mod, it is designed to allow complete replacements of how you code in df, such as external compilers. I have already have placing templates from an external script working, so I might be able to get your project to compile straight to df. It doesn't add anything not in recode, and should be compatible. I would be willing to add these features.

The problem is that Recode, has all of the socket work already done, plus other things that people who need to use blocks use. Its a obvious choice, and so I'm gonna use it. But, you mod, looks interesting, might contribute to it :heart:

**Pull request closed without merging**